src/caffe/layers/reshape_layer.cpp

```diff
@@ -0,0 +1,20 @@
+#include <vector>
+
+#include "caffe/common_layers.hpp"
+#include "caffe/layer.hpp"
+
+namespace caffe {
+
+template <typename Dtype>
+void ReshapeLayer<Dtype>::Reshape(const vector<Blob<Dtype>*>& bottom,
+      const vector<Blob<Dtype>*>& top) {
+  top[0]->Reshape(this->layer_param_.reshape_param().shape());
```

Comment:
@longjon I guess this should technically all be done in `LayerSetUp` since the top shape doesn't vary with the bottom shape?  (But then if special options were added when certain dims of the shape are set to 0 or -1, the top would then vary with the bottom shape, so it would be moved to `Reshape` then?  Maybe better to keep it in `Reshape` then?)